### PR TITLE
HTTP Validator API: slashing protection import and export

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**
migrates away from gRPC and implements a pure rest version of the following endpoints for the validator web

- [x] `/v2/validator/slashing-protection/export` GET
- [x] `/v2/validator/slashing-protection/import` POST

example run export
```
curl -X GET "http://localhost:7500/v2/validator/slashing-protection/export" -H "accept: application/json"
{"file":"{\n\t\"metadata\": {\n\t\t\"interchange_format_version\": \"5\",\n\t\t\"genesis_validators_root\": \"0x043db0d9a83813551ee2f33450d23797757d430911a9320530ad8a0eabc43efb\"\n\t},\n\t\"data\": [\n\t\t{\n\t\t\t\"pubkey\": \"0x855ae9c6184d6edd46351b375f16f541b2d33b0ed0da9be4571b13938588aee840ba606a946f0e8023ae3a4b2a43b4d4
```

**Which issues(s) does this PR fix?**

Part of #13041

**Other notes for review**

### CODE REVIEW

File: validator/rpc/handle_slashing.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// Steps:
+//  1. Call the function which exports the data from
+//     the validator's db into an EIP standard slashing protection format.
+//  2. Format and send JSON in the response.
+func (s *Server) ExportSlashingProtection(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ExportSlashingProtection")
+	defer span.End()
+
+	if s.valDB == nil {
+		http2.HandleError(w, "err finding validator database at path", http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handle_slashing.go:28]:
```suggestion
		http2.HandleError(w, "could not find validator database", http.StatusInternalServerError)
```
I changed the error message to something more standard. I don't think we ever return `err` in an error message and the `path` word is unclear. What path?

File: validator/rpc/handle_slashing.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// standard JSON string and inserts the data into validator DB.
+//
+// Read the JSON string passed through rpc, then call the func
+// which actually imports the data from the JSON file into our database. Use the Keymanager APIs if an API is required.
+func (s *Server) ImportSlashingProtection(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ImportSlashingProtection")
+	defer span.End()
+
+	if s.valDB == nil {
+		http2.HandleError(w, "err finding validator database at path", http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handle_slashing.go:58]:
```suggestion
		http2.HandleError(w, "could not find validator database", http.StatusInternalServerError)
```

File: validator/rpc/handle_slashing.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	if req.SlashingProtectionJson == "" {
+		http2.HandleError(w, "empty slashing_protection json specified", http.StatusBadRequest)
+		return
+	}
+	enc := []byte(req.SlashingProtectionJson)
+	buf := bytes.NewBuffer(enc)
+	if err := slashing.ImportStandardProtectionJSON(ctx, s.valDB, buf); err != nil {
+		http2.HandleError(w, err.Error(), http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handle_slashing.go:80]:
```suggestion
		http2.HandleError(w, errors.Wrap(err, "could not import slashing protection history").Error(), http.StatusInternalServerError)
```
For symmetry with the other endpoint

File: validator/rpc/handle_slashing.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		return
+	}
+	log.Info("Slashing protection JSON successfully imported")
```
Comment (rkapka) [inline validator/rpc/handle_slashing.go:83]: Do we really need this log? Client will receive a 200 response

File: validator/rpc/handle_slashing.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	case err == io.EOF:
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	case err != nil:
+		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if req.SlashingProtectionJson == "" {
+		http2.HandleError(w, "empty slashing_protection json specified", http.StatusBadRequest)
```
Comment (rkapka) [inline validator/rpc/handle_slashing.go:74]:
```suggestion
		http2.HandleError(w, "empty slashing_protection_json specified", http.StatusBadRequest)
```
Assuming you wanted the json field name here

File: validator/rpc/handle_slashing.go (hunk 4 above)
Comment (james-prysm) [reply validator/rpc/handle_slashing.go:83]: yeah i think it's safer for this in particular.

File: validator/rpc/handle_slashing.go
[the remark this answers is not in the document]
Comment (james-prysm) [reply validator/rpc/handle_slashing.go:1]: oops definitely missed that

**Review by rkapka — APPROVED** [review]

**Pull request merged**